plugin/core/types.py

```diff
@@ -65,25 +77,30 @@ def __init__(self,
         self.tcp_host = tcp_host
         self.tcp_mode = tcp_mode
         if not languages:
-            languages = [LanguageConfig(languageId, scopes, syntaxes)] if languageId else []
+            languages = [LanguageConfig(languageId, selector)] if languageId else []
         self.languages = languages
         self.enabled = enabled
         self.init_options = init_options
         self.settings = settings
         self.env = env
         self.experimental_capabilities = experimental_capabilities
 
+    def supports(self, base_scope: str) -> bool:
+        for language in self.languages:
+            if language.match(base_scope):
+                return True
+        return False
```

Comment:
I'm all for reducing vertical space usage. Can we use `any`?
```py
return any([language.match(base_scope) for language in self.languages])
```

Reply:
I suppose that's probably less efficient though as it needs to build the list first.

Reply:
@rchl in this case, I don't think efficiency would be an issue, I can't imagine that `self.languages` would ever be big enough to cause a performance impact.

Reply:
`return any([language.match(base_scope) for language in self.languages])`

such a wonderful one __line__ of code :)

Reply:
I believe one can simply omit the `[` and `]` to make it an anonymous lazy generator and not create a list, making it as "fast" as a for loop.